testsJVM3  javalib.lang.RuntimeTest.execDir failed

Ekrich reports that this works on his system.  I must have gotten a system
built on a Friday:

Clean directory,  Current X86_64 Linux, bad hair!

Reporting as a memory & place holder. 

``` 
testsJVM3/testOnly javalib.lang.RuntimeTest
[info] Test run started
[info] Test javalib.lang.RuntimeTest.execCommand started
[info] Test javalib.lang.RuntimeTest.execDir started
[error] Test javalib.lang.RuntimeTest.execDir failed: java.lang.AssertionError: expected:<Set(echo.sh, err.sh, hello.sh, ls)> but was:<Set(1)>, took 0.003 sec
[error]     at javalib.lang.RuntimeTest.execDir(RuntimeTest.scala:46)
[error]     ...
[info] Test javalib.lang.RuntimeTest.execEnvp started
[warn] Test assumption in test javalib.lang.RuntimeTest.execEnvp failed: org.junit.AssumptionViolatedException: Not compliant with JDK, took 0.003 sec
[info] Test run finished: 1 failed, 0 ignored, 3 total, 0.273s
[error] Failed: Total 3, Failed 1, Errors 0, Passed 1, Skipped 1
[error] Failed tests:
[error] 	javalib.lang.RuntimeTest
[error] (testsJVM3 / Test / testOnly) sbt.TestsFailedException: Tests unsuccessful

```

--- a/unit-tests/shared/src/test/scala/javalib/lang/RuntimeTest.scala
+++ b/unit-tests/shared/src/test/scala/javalib/lang/RuntimeTest.scala
@@ -14,11 +14,35 @@ import scala.scalanative.junit.utils.AssumesHelper._
 class RuntimeTest {
   import ProcessUtils._
   private val EOL = System.lineSeparator()
+
+  private val isLinux =
+    System.getProperty("os.name").toLowerCase.contains("linux")
+
+  /* See Issues #2649 & #2652 re use of 'ls'.
+   * Avoid a problem with ":.:", or variants in the user's PATH definition.
+   * Historically the 'resourceDir' has contained an executable file
+   * named 'ls'. When $PATH contains "." that file gets executed
+   * rather that the system standard "ls", causing the execDir test to
+   * fail.
+   *
+   * A private "ls" being found on the user's path before the
+   * system standard "ls" causes similar issues.
+   *
+   * Guessing and using a hard coded "/usr/bin/ls" is the least bad
+   * of several design alternatives ("dir", "find").
+   *
+   * The chosen location should work on unmodified Linux, macOS, and
+   * FreeBSD systems. Given the variety of system
+   * configurations in the wild, some OS or system is bound to have 'ls'
+   * in a different location.
+   */
+
   private def lsCommand =
-    if (isWindows)
+    if (isWindows) {
       Array("cmd", "/c", "dir", "/b")
-    else
-      Array("ls")
+    } else {
+      Array("/usr/bin/ls") // See /* */ block comment immediately above.
+    }
 
   @Test def execCommand(): Unit = {
     val proc = Runtime.getRuntime.exec(lsCommand :+ resourceDir)

--- a/unit-tests/shared/src/test/scala/javalib/lang/RuntimeTest.scala
+++ b/unit-tests/shared/src/test/scala/javalib/lang/RuntimeTest.scala
@@ -40,10 +40,8 @@ class RuntimeTest {
   private def lsCommand =
     if (isWindows) {
       Array("cmd", "/c", "dir", "/b")
-    } else if (isMacOs || isFreeBSD) {
+    } else {
       Array("/bin/ls")
-    } else { // Linux
-      Array("/usr/bin/ls")
     }
 
   @Test def execCommand(): Unit = {

--- a/unit-tests/shared/src/test/scala/javalib/lang/RuntimeTest.scala
+++ b/unit-tests/shared/src/test/scala/javalib/lang/RuntimeTest.scala
@@ -28,10 +28,10 @@ class RuntimeTest {
    * A private "ls" being found on the user's path before the
    * system standard "ls" causes similar issues.
    *
-   * Guessing and using a hard coded "/usr/bin/ls" is the least bad
+   * Guessing and using a hard coded "ls" is the least bad
    * of several design alternatives ("dir", "find").
    *
-   * The chosen location should work on unmodified Linux, macOS, and
+   * The chosen locations should work on unmodified Linux, macOS, and
    * FreeBSD systems. Given the variety of system
    * configurations in the wild, some OS or system is bound to have 'ls'
    * in a different location.
@@ -40,8 +40,10 @@ class RuntimeTest {
   private def lsCommand =
     if (isWindows) {
       Array("cmd", "/c", "dir", "/b")
-    } else {
-      Array("/usr/bin/ls") // See /* */ block comment immediately above.
+    } else if (isMacOs || isFreeBSD) {
+      Array("/bin/ls")
+    } else { // Linux
+      Array("/usr/bin/ls")
     }
 
   @Test def execCommand(): Unit = {

--- a/unit-tests/shared/src/test/scala/javalib/lang/RuntimeTest.scala
+++ b/unit-tests/shared/src/test/scala/javalib/lang/RuntimeTest.scala
@@ -15,9 +15,6 @@ class RuntimeTest {
   import ProcessUtils._
   private val EOL = System.lineSeparator()
 
-  private val isLinux =
-    System.getProperty("os.name").toLowerCase.contains("linux")
-
   /* See Issues #2649 & #2652 re use of 'ls'.
    * Avoid a problem with ":.:", or variants in the user's PATH definition.
    * Historically the 'resourceDir' has contained an executable file

